storage/src/main/java/io/delta/storage/internal/S3LogStoreUtil.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * Copyright (2022) The Delta Lake Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.delta.storage.internal;
+
+import com.amazonaws.services.s3.model.ListObjectsV2Request;
+import org.apache.hadoop.fs.*;
+import org.apache.hadoop.fs.s3a.*;
+
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.util.HashSet;
+
+import static org.apache.hadoop.fs.s3a.Constants.DEFAULT_MAX_PAGING_KEYS;
+import static org.apache.hadoop.fs.s3a.Constants.MAX_PAGING_KEYS;
+import static org.apache.hadoop.fs.s3a.S3AUtils.iteratorToStatuses;
+
+
+/**
+ * Static utility methods for the S3SingleDriverLogStore.
+ *
+ * Used to trick the class loader so we can use methods of org.apache.hadoop:hadoop-aws without needing to load this as
+ * a dependency for tests in core.
+ */
+public final class S3LogStoreUtil {
+    private S3LogStoreUtil() {}
+
+    private static PathFilter ACCEPT_ALL = new PathFilter() {
+        @Override
+        public boolean accept(Path file) {
+            return true;
+        }
+
+        @Override
+        public String toString() {
+            return "ACCEPT_ALL";
+        }
+    };
+
+    /**
+     * Uses the S3ListRequest.v2 interface with the startAfter parameter to only list files
+     * which are lexicographically greater than resolvedPath.
+     */
+    private static RemoteIterator<S3AFileStatus> s3ListFrom(S3AFileSystem s3afs, Path resolvedPath, Path parentPath) throws IOException {
```

Comment:
can you format this like so
```java
private static RemoteIterator<S3AFileStatus> s3ListFrom(
        S3AFileSystem s3afs,
        Path resolvedPath,
        Path parentPath) throws IOException {
    // code starts here
```

Reply:
Done in https://github.com/delta-io/delta/pull/1210/commits/2a01945be7b9980f621d226f23344c4962a917f6